tests/tasty/Arbitrary.hs

```diff
@@ -116,7 +149,7 @@ arbitrarySort 0 = oneof (pure <$> [FInt, FReal, FNum, FFrac])
 arbitrarySort n = frequency
   [ (1, FObj <$> arbitrary)
   , (1, FVar <$> arbitrary)
-  , (1, FFunc <$> arbitrarySort' <*> arbitrarySort')
+  -- , (1, FFunc <$> arbitrarySort' <*> arbitrarySort')
```

Comment:
@facundominguez this is the line that was causing the strange errors.  The problem is with a function called `unifySorts` in liquid-fixpoint.  Basically, it calls `die`, but I have no idea how to fix it so I just removed this case from the sorts.  I'm not sure why the sorts are trying to be unified in any case, since we are testing a substitution, not type checking...

Reply:
Sometimes `inlineInExpr` would need to insert sort coercions when inlining a binding which doesn't have exactly the same sort as expected at the place in which the binding is referenced. Here `inlineInExpr` needs to determine what the actual and the expected sorts are, and there is where sort inference is involved, which sometimes seems to require unification.

Are you sure no other code relies on the case that you are leaving out? If you are, then this is fine. But otherwise, it would be preferable to just filter this case at the places where you use it.

Also, maybe arbitrary sorts are not necessary and a single sort could be used, say `FInt`.

Reply:
I like the idea of using a single sort, actually, because that would make the test cases much smaller to read.